bfd/coff-aarch64.c

```diff
@@ -523,11 +523,30 @@ coff_pe_aarch64_relocate_section (bfd *output_bfd,
 
       h = obj_coff_sym_hashes (input_bfd)[symndx];
 
-      if (h && h->root.type == bfd_link_hash_defined)
+      if (h && (h->root.type == bfd_link_hash_defined
+	|| h->root.type == bfd_link_hash_defweak))
 	{
 	  sec = h->root.u.def.section;
 	  sym_value = h->root.u.def.value;
 	}
+      else if (h && h->root.type == bfd_link_hash_undefweak
+	&& h->symbol_class == C_NT_WEAK && h->numaux == 1)
+	{
+	  struct coff_link_hash_entry *h2 =
```

Comment:
As this block of code is identical to the code in `cofflink.c`, I'd suggest to extract it into a shared function.

Reply:
Instead it would be good to merge this function with _bfd_coff_generic_relocate_section. However it requires extra time and it is low priority for now. I suggest to keep it like this until the proper refactoring.